Add DNS 2017-10-01 API version

nkindberg opened the pull request:
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

Commit:
Add DNS 2017-10-01 API version

AutorestCI commented:
# Automation for azure-sdk-for-go

nkindberg commented:
Please diff against DNS 2017-09-01 to see the set of changes. The autorest errors related to camel-case are existing errors that we need to continue to carry over for now.

AutorestCI commented:
# Automation for azure-sdk-for-python

mcardosos commented:
@nkindberg I guess this API version is based on `2017-09-01`? Please submit your PR this way:
- FIrst commit: copy and paste swaggers on which new API version is based on to the new API folder
- Second commit: modify swaggers so they reflect the new API

This is the easiest way for me to look at the diff, so faster review.

Commit:
Copy 2017-09-01 DNS API version to 2017-10-01

Commit:
Add 2017-10-01 DNS API version

nkindberg commented:
@mcardosos done. Compare 2nd commit (2017-09-01) against 3rd commit (2017-10-01)

mcardosos commented:
@nkindberg Fantastic! Thanks :D

mcardosos commented on specification/dns/resource-manager/Microsoft.Network/stable/2017-10-01/dns.json:
```diff
@@ -0,0 +1,1584 @@
+            "description": "The maximum number of record sets to return. If not specified, returns up to 100 record sets."
+          },
+          {
+            "name": "$recordsetnamesuffix",
```
>$recordsetnamesuffix [](start = 21, length = 20)

For this param in particular it would be a good idea to use x-ms-client-name extension

nkindberg replied:
Yes, this is a good suggestion. I am going to fix it for this new API "/all". It looks like somehow 2 existing APIs slipped through the cracks with this same parameter without this unfortunately. I am going to avoid making a breaking SDK change on the existing 2 for now, unless you feel there is a compelling case here for a breaking SDK change.

mcardosos replied:
Oh no, lets not create breaking changes just for changing a name :)

mcardosos commented on specification/dns/resource-manager/Microsoft.Network/stable/2017-10-01/dns.json:
```diff
@@ -0,0 +1,1584 @@
+        "tags": [
+          "RecordSets"
+        ],
+        "operationId": "RecordSets_ListAllByDnsZone",
```
>RecordSets_ListAllByDnsZone [](start = 24, length = 27)

Could x-ms-odata extension be useful here?

nkindberg replied:
Doesn't look like it. The autorest generator for the C# SDK fails if I add that due to not having $filter parameter, which we do not support in this API.

FATAL: System.InvalidOperationException: Method with x-ms-odata needs to have "$filter" parameter.

mcardosos replied:
Okok

mcardosos commented:
@nkindberg Am I right assuming this API has already been deployed?

mcardosos reviewed (changes requested):
This looks pretty good. Added just two comments.
Could you explain a bit more on why camel case errors should be ignored?
There is also another [linter error](https://travis-ci.org/Azure/azure-rest-api-specs/jobs/347583340#L825-L836), unrelated to camel case stuff.

Commit:
Add x-ms-client-name for recordsetnamesuffix for RecordSets_ListAllByDnsZone

nkindberg commented:
@mcardosos addressed the comments.

This API version is indeed already deployed service-side.

The camel case errors need to be ignored for now because they require significant service-side changes. We are planning to address them in a future API version, but didn't have the budget for it in this version.

The other linter error about OperationsAPIImplementation is an expected error for us due to how the Microsoft.Network resource provider works. Operations API is implemented as part of the Microsoft.Network Swagger/SDK. DNS is essentially a sub-RP of Microsoft.Network and therefore not serving this API.

mcardosos commented:
Good :)
For the linter errors, consider [requesting supression](https://github.com/Azure/adx-documentation-pr/wiki/Swagger-Validation-Errors-Suppression)

ravbhatnagar commented:
Signing off from ARM.

mcardosos reviewed (approved):
Great! Super smooth
@nkindberg Please submit the suppressions in a different PR :)

AutorestCI commented:
# Automation for azure-sdk-for-node
